### PR TITLE
libzip: Build a bottle for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
```
failed: brew linkage --test libzip

        System libraries:
  /lib/x86_64-linux-gnu/libbz2.so.1.0
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libcrypto.so.1.0.0
No broken dylib links
Unwanted system libraries:
  libbz2.so.1.0
  libcrypto.so.1.0.0
```

**Comment by iMichka** [comment]
@sjackman @maxim-belkin Do we have formulae for these? Looks like we don't.

**Comment by maxim-belkin** [comment]
`bzip2`

**Comment by maxim-belkin** [comment]
and (I think) `openssl`

**Comment by maxim-belkin** [comment]
yes: `/home/linuxbrew/.linuxbrew/Cellar/openssl/1.0.2n_1/lib/libcrypto.so.1.0.0`

**Comment by sjackman** [comment]
The linkage to `openssl` is a bit unexpected and weird. On macOS it depends on only on `libz.1.dylib`.
```
❯❯❯ brew linkage libzip                          ⏎ master ✱ ◼
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /usr/local/Cellar/libzip/1.1.2/lib/libzip.4.dylib (libzip)
```

File: Formula/libzip.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@ class Libzip < Formula
 
   conflicts_with "libtcod", :because => "both install `zip.h` header"
 
-  depends_on "zlib" unless OS.mac?
+  unless OS.mac?
+    depends_on "bzip2"
+    depends_on "zlib"
+    depends_on "openssl"
+  end
```
Comment (maxim-belkin) [inline Formula/libzip.rb]: + `depends_on "zip" => :test` ?

Comment (iMichka) [reply Formula/libzip.rb]: Is this provided in brew 1.5.8? And I think brew test bot has some patches too. So we may start using this at a later point. I fixed the test like we did for the `unzip` test.

**Comment by iMichka** [comment]
>  The linkage to openssl is a bit unexpected and weird

I have seen many packages building different stuff on mac and on linux that nothing surprises me anymore.

**Pull request closed without merging**